packages/chevrotain/tsconfig.esm.json

```diff
@@ -0,0 +1,9 @@
+{
+    "extends": "./tsconfig.json",
+    "compilerOptions": {
+        "moduleResolution": "node",
+        "module": "esnext",
```

Comment:
Do we need `esnext` module or just `es2015`?

Reply:
The only difference between “esnext” and “es2015” is that “esnext” supports new features such as dynamic import. 

Though these new features are not used in this project currently, I think setting module option to “esnext” is better in case we need the new import features in the future. 

> https://github.com/microsoft/TypeScript/issues/24082#issuecomment-388590629

https://github.com/microsoft/TypeScript/blob/d982014d733445f478b5cad8a938d2d2a56f1080/src/compiler/diagnosticMessages.json#L910:L913